File: recovered/src/com/lumiyaviewer/lumiya/render/BoundingBox.java
package com.lumiyaviewer.lumiya.render;

import android.opengl.GLES20;
import android.opengl.GLES30;
import android.support.v4.view.InputDeviceCompat;
import com.lumiyaviewer.lumiya.render.glres.GLQuery;
import com.lumiyaviewer.lumiya.render.glres.buffers.GLLoadableBuffer;
import com.lumiyaviewer.lumiya.render.glres.buffers.GLVertexArrayObject;
import com.lumiyaviewer.rawbuffers.DirectByteBuffer;

/* loaded from: classes.dex */
public class BoundingBox {
    private static final float cubeSize = 0.5f;
    private static float[] cubeVertices = {-0.5f, -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f};
    private final GLVertexArrayObject vertexArrayObject;
    private final GLLoadableBuffer vertexBuffer;

    public BoundingBox(RenderContext renderContext) {
        DirectByteBuffer directByteBuffer = new DirectByteBuffer(cubeVertices.length * 4);
        directByteBuffer.loadFromFloatArray(0, cubeVertices, 0, cubeVertices.length);
        this.vertexBuffer = new GLLoadableBuffer(directByteBuffer);
        this.vertexArrayObject = new GLVertexArrayObject(renderContext.glResourceManager, 1);
        GLES20.glUseProgram(renderContext.boundingBoxProgram.getHandle());
        this.vertexArrayObject.Bind(0);
        this.vertexBuffer.Bind20(renderContext, renderContext.boundingBoxProgram.vPosition, 3, 5126, 12, 0);
        GLES20.glBindBuffer(34963, 0);
        this.vertexArrayObject.Unbind();
        GLES20.glUseProgram(0);
    }

    public static void EndOcclusionQueries(RenderContext renderContext) {
        GLES20.glUseProgram(0);
        renderContext.curPrimProgram = null;
        GLES30.glColorMask(true, true, true, true);
        GLES30.glDepthMask(true);
        GLES30.glDepthFunc(InputDeviceCompat.SOURCE_DPAD);
        GLES30.glEnable(2884);
    }

    public static void PrepareOcclusionQueries(RenderContext renderContext) {
        GLES30.glColorMask(false, false, false, false);
        GLES30.glDepthMask(false);
        GLES30.glDisable(2884);
        GLES30.glDepthFunc(515);
        GLES20.glUseProgram(renderContext.boundingBoxProgram.getHandle());
        renderContext.glModelApplyMatrix(renderContext.boundingBoxProgram.uMVPMatrix);
    }

    public void OcclusionQuery(RenderContext renderContext, GLQuery gLQuery) {
        renderContext.glObjWorldApplyMatrix(renderContext.boundingBoxProgram.uObjWorldMatrix);
        renderContext.glObjScaleApplyVector(renderContext.boundingBoxProgram.uObjCoordScale);
        this.vertexArrayObject.Bind(0);
        gLQuery.BeginOcclusionQuery(renderContext);
        GLES20.glDrawArrays(4, 0, cubeVertices.length / 3);
        gLQuery.EndOcclusionQuery();
        this.vertexArrayObject.Unbind();
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/BoundingBox.kt
package com.lumiyaviewer.lumiya.render

import android.opengl.GLES20
import android.opengl.GLES30
import com.lumiyaviewer.lumiya.render.glres.GLQuery
import com.lumiyaviewer.lumiya.render.glres.buffers.GLLoadableBuffer
import com.lumiyaviewer.lumiya.render.glres.buffers.GLVertexArrayObject
import com.lumiyaviewer.rawbuffers.DirectByteBuffer

class BoundingBox(renderContext: RenderContext) {
    private val vertexArrayObject: GLVertexArrayObject
    private val vertexBuffer: GLLoadableBuffer

    init {
        val directByteBuffer = DirectByteBuffer(cubeVertices.size * 4)
        directByteBuffer.loadFromFloatArray(0, cubeVertices, 0, cubeVertices.size)
        vertexBuffer = GLLoadableBuffer(directByteBuffer)
        vertexArrayObject = GLVertexArrayObject(renderContext.glResourceManager, 1)
        GLES20.glUseProgram(renderContext.boundingBoxProgram.getHandle())
        vertexArrayObject.Bind(0)
        vertexBuffer.Bind20(renderContext, renderContext.boundingBoxProgram.vPosition, 3, GLES20.GL_FLOAT, 12, 0)
        GLES20.glBindBuffer(GLES20.GL_ELEMENT_ARRAY_BUFFER, 0)
        vertexArrayObject.Unbind()
        GLES20.glUseProgram(0)
    }

    fun OcclusionQuery(renderContext: RenderContext, glQuery: GLQuery) {
        renderContext.glObjWorldApplyMatrix(renderContext.boundingBoxProgram.uObjWorldMatrix)
        renderContext.glObjScaleApplyVector(renderContext.boundingBoxProgram.uObjCoordScale)
        vertexArrayObject.Bind(0)
        glQuery.BeginOcclusionQuery(renderContext)
        GLES20.glDrawArrays(GLES20.GL_TRIANGLES, 0, cubeVertices.size / 3)
        glQuery.EndOcclusionQuery()
        vertexArrayObject.Unbind()
    }

    companion object {
        private val cubeVertices = floatArrayOf(
            -0.5f, -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f,
            -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f,
            0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f, -0.5f, -0.5f,
            -0.5f, -0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f, -0.5f, 0.5f,
            -0.5f, -0.5f, 0.5f, -0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f,
            -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f,
            -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, -0.5f,
            0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, 0.5f, -0.5f, -0.5f, 0.5f,
            -0.5f, 0.5f, 0.5f, 0.5f, -0.5f, 0.5f, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f, 0.5f,
            0.5f, -0.5f, 0.5f, -0.5f, 0.5f, 0.5f
        )

        @JvmStatic
        fun EndOcclusionQueries(renderContext: RenderContext) {
            GLES20.glUseProgram(0)
            renderContext.curPrimProgram = null
            GLES30.glColorMask(true, true, true, true)
            GLES30.glDepthMask(true)
            GLES30.glDepthFunc(GLES30.GL_LESS)
            GLES30.glEnable(GLES30.GL_DEPTH_TEST)
        }

        @JvmStatic
        fun PrepareOcclusionQueries(renderContext: RenderContext) {
            GLES30.glColorMask(false, false, false, false)
            GLES30.glDepthMask(false)
            GLES30.glDisable(GLES30.GL_DEPTH_TEST)
            GLES30.glDepthFunc(GLES30.GL_LEQUAL)
            GLES20.glUseProgram(renderContext.boundingBoxProgram.getHandle())
            renderContext.glModelApplyMatrix(renderContext.boundingBoxProgram.uMVPMatrix)
        }
    }
}
